diff --git a/src/Lazy/LazyUuidFromString.php b/src/Lazy/LazyUuidFromString.php
index 8d3129c..afbc94d 100644
--- a/src/Lazy/LazyUuidFromString.php
+++ b/src/Lazy/LazyUuidFromString.php
@@ -16,7 +16,6 @@ namespace Ramsey\Uuid\Lazy;
 
 use DateTimeInterface;
 use Ramsey\Uuid\Converter\NumberConverterInterface;
-use Ramsey\Uuid\Exception\UnsupportedOperationException;
 use Ramsey\Uuid\Fields\FieldsInterface;
 use Ramsey\Uuid\Rfc4122\UuidV1;
 use Ramsey\Uuid\Rfc4122\UuidV6;
@@ -117,7 +116,7 @@ final class LazyUuidFromString implements UuidInterface
 
     public function getNumberConverter(): NumberConverterInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getNumberConverter();
+        return $this->unwrapIfRequired()->getNumberConverter();
     }
 
     /**
@@ -125,82 +124,82 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getFieldsHex(): array
     {
-        return ($this->unwrapped ?? $this->unwrap())->getFieldsHex();
+        return $this->unwrapIfRequired()->getFieldsHex();
     }
 
     public function getClockSeqHiAndReservedHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSeqHiAndReservedHex();
+        return $this->unwrapIfRequired()->getClockSeqHiAndReservedHex();
     }
 
     public function getClockSeqLowHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSeqLowHex();
+        return $this->unwrapIfRequired()->getClockSeqLowHex();
     }
 
     public function getClockSequenceHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSequenceHex();
+        return $this->unwrapIfRequired()->getClockSequenceHex();
     }
 
     public function getDateTime(): DateTimeInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getDateTime();
+        return $this->unwrapIfRequired()->getDateTime();
     }
 
     public function getLeastSignificantBitsHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getLeastSignificantBitsHex();
+        return $this->unwrapIfRequired()->getLeastSignificantBitsHex();
     }
 
     public function getMostSignificantBitsHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getMostSignificantBitsHex();
+        return $this->unwrapIfRequired()->getMostSignificantBitsHex();
     }
 
     public function getNodeHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getNodeHex();
+        return $this->unwrapIfRequired()->getNodeHex();
     }
 
     public function getTimeHiAndVersionHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeHiAndVersionHex();
+        return $this->unwrapIfRequired()->getTimeHiAndVersionHex();
     }
 
     public function getTimeLowHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeLowHex();
+        return $this->unwrapIfRequired()->getTimeLowHex();
     }
 
     public function getTimeMidHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeMidHex();
+        return $this->unwrapIfRequired()->getTimeMidHex();
     }
 
     public function getTimestampHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimestampHex();
+        return $this->unwrapIfRequired()->getTimestampHex();
     }
 
     public function getUrn(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getUrn();
+        return $this->unwrapIfRequired()->getUrn();
     }
 
     public function getVariant(): ?int
     {
-        return ($this->unwrapped ?? $this->unwrap())->getVariant();
+        return $this->unwrapIfRequired()->getVariant();
     }
 
     public function getVersion(): ?int
     {
-        return ($this->unwrapped ?? $this->unwrap())->getVersion();
+        return $this->unwrapIfRequired()->getVersion();
     }
 
     public function compareTo(UuidInterface $other): int
     {
-        return ($this->unwrapped ?? $this->unwrap())->compareTo($other);
+        return $this->unwrapIfRequired()->compareTo($other);
     }
 
     public function equals(?object $other): bool
@@ -223,17 +222,17 @@ final class LazyUuidFromString implements UuidInterface
 
     public function getFields(): FieldsInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getFields();
+        return $this->unwrapIfRequired()->getFields();
     }
 
     public function getHex(): Hexadecimal
     {
-        return ($this->unwrapped ?? $this->unwrap())->getHex();
+        return $this->unwrapIfRequired()->getHex();
     }
 
     public function getInteger(): IntegerObject
     {
-        return ($this->unwrapped ?? $this->unwrap())->getInteger();
+        return $this->unwrapIfRequired()->getInteger();
     }
 
     public function toString(): string
@@ -258,12 +257,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSeqHiAndReserved(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeqHiAndReserved()->toString());
+        return $this->unwrapIfRequired()->getClockSeqHiAndReserved();
     }
 
     /**
@@ -273,12 +267,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSeqLow(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeqLow()->toString());
+        return $this->unwrapIfRequired()->getClockSeqLow();
     }
 
     /**
@@ -288,12 +277,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSequence(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeq()->toString());
+        return $this->unwrapIfRequired()->getClockSequence();
     }
 
     /**
@@ -302,9 +286,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getLeastSignificantBits(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        return $instance->getNumberConverter()->fromHex(substr($instance->getHex()->toString(), 16));
+        return $this->unwrapIfRequired()->getLeastSignificantBits();
     }
 
     /**
@@ -313,9 +295,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getMostSignificantBits(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        return $instance->getNumberConverter()->fromHex(substr($instance->getHex()->toString(), 0, 16));
+        return $this->unwrapIfRequired()->getMostSignificantBits();
     }
 
     /**
@@ -325,12 +305,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getNode(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getNode()->toString());
+        return $this->unwrapIfRequired()->getNode();
     }
 
     /**
@@ -340,12 +315,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeHiAndVersion(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeHiAndVersion()->toString());
+        return $this->unwrapIfRequired()->getTimeHiAndVersion();
     }
 
     /**
@@ -355,12 +325,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeLow(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeLow()->toString());
+        return $this->unwrapIfRequired()->getTimeLow();
     }
 
     /**
@@ -370,12 +335,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeMid(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeMid()->toString());
+        return $this->unwrapIfRequired()->getTimeMid();
     }
 
     /**
@@ -385,21 +345,12 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimestamp(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        if ($fields->getVersion() !== 1) {
-            throw new UnsupportedOperationException('Not a time-based UUID');
-        }
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimestamp()->toString());
+        return $this->unwrapIfRequired()->getTimestamp();
     }
 
     public function toUuidV1(): UuidV1
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
+        $instance = $this->unwrapIfRequired();
 
         if ($instance instanceof UuidV1) {
             return $instance;
@@ -412,13 +363,18 @@ final class LazyUuidFromString implements UuidInterface
 
     public function toUuidV6(): UuidV6
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
+        $instance = $this->unwrapIfRequired();
 
         assert($instance instanceof UuidV6);
 
         return $instance;
     }
 
+    private function unwrapIfRequired(): UuidInterface
+    {
+        return $this->unwrapped ?? $this->unwrap();
+    }
+
     private function unwrap(): UuidInterface
     {
         return $this->unwrapped = (new UuidFactory())->fromString($this->uuid);
