diff --git a/src/Lazy/LazyUuidFromString.php b/src/Lazy/LazyUuidFromString.php
index 8d3129c..d0d6807 100644
--- a/src/Lazy/LazyUuidFromString.php
+++ b/src/Lazy/LazyUuidFromString.php
@@ -22,6 +22,7 @@ use Ramsey\Uuid\Rfc4122\UuidV1;
 use Ramsey\Uuid\Rfc4122\UuidV6;
 use Ramsey\Uuid\Type\Hexadecimal;
 use Ramsey\Uuid\Type\Integer as IntegerObject;
+use Ramsey\Uuid\Uuid;
 use Ramsey\Uuid\UuidFactory;
 use Ramsey\Uuid\UuidInterface;
 use ValueError;
@@ -117,7 +118,7 @@ final class LazyUuidFromString implements UuidInterface
 
     public function getNumberConverter(): NumberConverterInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getNumberConverter();
+        return $this->requireFullUuid()->getNumberConverter();
     }
 
     /**
@@ -125,82 +126,82 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getFieldsHex(): array
     {
-        return ($this->unwrapped ?? $this->unwrap())->getFieldsHex();
+        return $this->requireFullUuid()->getFieldsHex();
     }
 
     public function getClockSeqHiAndReservedHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSeqHiAndReservedHex();
+        return $this->requireFullUuid()->getClockSeqHiAndReservedHex();
     }
 
     public function getClockSeqLowHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSeqLowHex();
+        return $this->requireFullUuid()->getClockSeqLowHex();
     }
 
     public function getClockSequenceHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getClockSequenceHex();
+        return $this->requireFullUuid()->getClockSequenceHex();
     }
 
     public function getDateTime(): DateTimeInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getDateTime();
+        return $this->requireFullUuid()->getDateTime();
     }
 
     public function getLeastSignificantBitsHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getLeastSignificantBitsHex();
+        return $this->requireFullUuid()->getLeastSignificantBitsHex();
     }
 
     public function getMostSignificantBitsHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getMostSignificantBitsHex();
+        return $this->requireFullUuid()->getMostSignificantBitsHex();
     }
 
     public function getNodeHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getNodeHex();
+        return $this->requireFullUuid()->getNodeHex();
     }
 
     public function getTimeHiAndVersionHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeHiAndVersionHex();
+        return $this->requireFullUuid()->getTimeHiAndVersionHex();
     }
 
     public function getTimeLowHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeLowHex();
+        return $this->requireFullUuid()->getTimeLowHex();
     }
 
     public function getTimeMidHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimeMidHex();
+        return $this->requireFullUuid()->getTimeMidHex();
     }
 
     public function getTimestampHex(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getTimestampHex();
+        return $this->requireFullUuid()->getTimestampHex();
     }
 
     public function getUrn(): string
     {
-        return ($this->unwrapped ?? $this->unwrap())->getUrn();
+        return $this->requireFullUuid()->getUrn();
     }
 
     public function getVariant(): ?int
     {
-        return ($this->unwrapped ?? $this->unwrap())->getVariant();
+        return $this->requireFullUuid()->getVariant();
     }
 
     public function getVersion(): ?int
     {
-        return ($this->unwrapped ?? $this->unwrap())->getVersion();
+        return $this->requireFullUuid()->getVersion();
     }
 
     public function compareTo(UuidInterface $other): int
     {
-        return ($this->unwrapped ?? $this->unwrap())->compareTo($other);
+        return $this->requireFullUuid()->compareTo($other);
     }
 
     public function equals(?object $other): bool
@@ -223,17 +224,17 @@ final class LazyUuidFromString implements UuidInterface
 
     public function getFields(): FieldsInterface
     {
-        return ($this->unwrapped ?? $this->unwrap())->getFields();
+        return $this->requireFullUuid()->getFields();
     }
 
     public function getHex(): Hexadecimal
     {
-        return ($this->unwrapped ?? $this->unwrap())->getHex();
+        return $this->requireFullUuid()->getHex();
     }
 
     public function getInteger(): IntegerObject
     {
-        return ($this->unwrapped ?? $this->unwrap())->getInteger();
+        return $this->requireFullUuid()->getInteger();
     }
 
     public function toString(): string
@@ -258,12 +259,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSeqHiAndReserved(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeqHiAndReserved()->toString());
+        return $this->requireFullUuid()->getClockSeqHiAndReserved();
     }
 
     /**
@@ -273,12 +269,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSeqLow(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeqLow()->toString());
+        return $this->requireFullUuid()->getClockSeqLow();
     }
 
     /**
@@ -288,12 +279,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getClockSequence(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getClockSeq()->toString());
+        return $this->requireFullUuid()->getClockSequence();
     }
 
     /**
@@ -302,9 +288,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getLeastSignificantBits(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        return $instance->getNumberConverter()->fromHex(substr($instance->getHex()->toString(), 16));
+        return $this->requireFullUuid()->getLeastSignificantBits();
     }
 
     /**
@@ -313,9 +297,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getMostSignificantBits(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        return $instance->getNumberConverter()->fromHex(substr($instance->getHex()->toString(), 0, 16));
+        return $this->requireFullUuid()->getMostSignificantBits();
     }
 
     /**
@@ -325,12 +307,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getNode(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getNode()->toString());
+        return $this->requireFullUuid()->getNode();
     }
 
     /**
@@ -340,12 +317,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeHiAndVersion(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeHiAndVersion()->toString());
+        return $this->requireFullUuid()->getTimeHiAndVersion();
     }
 
     /**
@@ -355,12 +327,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeLow(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeLow()->toString());
+        return $this->requireFullUuid()->getTimeLow();
     }
 
     /**
@@ -370,12 +337,7 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimeMid(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimeMid()->toString());
+        return $this->requireFullUuid()->getTimeMid();
     }
 
     /**
@@ -385,21 +347,12 @@ final class LazyUuidFromString implements UuidInterface
      */
     public function getTimestamp(): string
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
-
-        $fields = $instance->getFields();
-        assert($fields instanceof \Ramsey\Uuid\Rfc4122\FieldsInterface);
-
-        if ($fields->getVersion() !== 1) {
-            throw new UnsupportedOperationException('Not a time-based UUID');
-        }
-
-        return $instance->getNumberConverter()->fromHex($fields->getTimestamp()->toString());
+        return $this->requireFullUuid()->getTimestamp();
     }
 
     public function toUuidV1(): UuidV1
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
+        $instance = $this->requireFullUuid();
 
         if ($instance instanceof UuidV1) {
             return $instance;
@@ -412,13 +365,25 @@ final class LazyUuidFromString implements UuidInterface
 
     public function toUuidV6(): UuidV6
     {
-        $instance = ($this->unwrapped ?? $this->unwrap());
+        $instance = $this->requireFullUuid();
 
         assert($instance instanceof UuidV6);
 
         return $instance;
     }
 
+    /**
+     * Unwraps the UUID and returns the full Uuid instance.
+     */
+    private function requireFullUuid(): Uuid
+    {
+        $instance = ($this->unwrapped ?? $this->unwrap());
+
+        assert($instance instanceof Uuid);
+
+        return $instance;
+    }
+
     private function unwrap(): UuidInterface
     {
         return $this->unwrapped = (new UuidFactory())->fromString($this->uuid);
